### PR TITLE
ReleaseNote: Mention SpecialCaseList change

### PR DESCRIPTION
**Pull request by MaskRay**

### CODE REVIEW

**Comment by tstellar** [comment]
Is this a change from LLVM 17 to LLVM 18 or from LLVM 18.0.3 to 18.0.4 ?

**Comment by pogo59** [comment]
@tstellar The PR was committed in December, so this is a change from LLVM17 to LLVM18. It is a behavior change that is tripping up some people, and I asked @MaskRay for a release note.

If it's too late for that, I can propagate his release note to our downstream release.

**Comment by ellishg** [comment]
Should we add a link to the docs?

https://clang.llvm.org/docs/SanitizerSpecialCaseList.html#format

**Comment by MaskRay** [comment]
> Should we add a link to the docs?
> 
> [clang.llvm.org/docs/SanitizerSpecialCaseList.html#format](https://clang.llvm.org/docs/SanitizerSpecialCaseList.html#format)

Thanks for the suggestion. Added

File: llvm/docs/ReleaseNotes.rst [hunk 1]
```diff
@@ -435,6 +435,12 @@ Changes to Sanitizers
 ---------------------
 * HWASan now defaults to detecting use-after-scope bugs.
 
+* `SpecialCaseList <https://clang.llvm.org/docs/SanitizerSpecialCaseList.html#format>`_
+  used by sanitizer ignore lists (e.g. ``*_ignorelist.txt`` in the Clang
+  resource directory) now uses glob patterns instead of a variant of POSIX
+  Extended Regular Expression (where ``*`` is translated to ``.*``) by default.
+  Search for ``|`` to find patterns that may have different meanings now.
```
Comment (pogo59) [inline llvm/docs/ReleaseNotes.rst:442]:
```suggestion
  Search for ``|`` to find patterns that may have different meanings now, and
  replace ``a|b`` with ``{a,b}``.
```

**Review by ellishg — APPROVED** [review]

**Pull request merged**